dynamicscrm/metadata.go

```diff
@@ -34,109 +34,181 @@ func (c *Connector) listObjectMetadata(
 		return nil, common.ErrMissingObjects
 	}
 
-	url, err := c.getURL("$metadata")
+	// Collect display name for the Object and its fields (name+display)
+	result := map[string]common.ObjectMetadata{}
+
+	for _, objectName := range objectNames {
+		objectDisplayName, err := c.metadataGetObjectDisplayName(ctx, objectName)
+		if err != nil {
+			return nil, err
+		}
+
+		fields, err := c.metadataGetFields(ctx, objectName)
```

Comment:
Likewise, we could skip the `metadata` prefix and call this `getFieldsForObject` or something instead